### ISSUE TITLE
SC2086: Should recognize that `${PARAM+WORD}` is similar to `${PARAM:+WORD}`

### ISSUE THREAD

**Issue by LukeShu** [koalaman/shellcheck#883]
#### For bugs
- Rule Id: 2086
- My shellcheck version: 0.4.6, online
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

Shellcheck currently correctly recognizes that for `${PARAM:+WORD}` that the outer expression doesn't need to be quoted, because it is the WORD expression that it will evaluate to.  However, it does not recognize the very similar `${PARAM+WORD}` expression format (no `:`).  They aren't quite the same (`:+` is roughly `if (unset(var) || value(var) == "")`, while `+` is roughly `if (unset(var))`); however for the purpose of deciding whether to trigger SC2086, they are the same.

If there are other checks that involve `${PARAM:-WORD}`, `${PARAM:=WORD}`, `${PARAM:?WORD}`, or `${PARAM:+WORD}`, I suspect that they also lack support for the colon-les version of the expression.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
ary=(a b c d)
printf '%s\n' ${ary+"${ary[@]}"}
```

#### Here's what shellcheck currently says:

```
Line 3:
printf '%s\n' ${ary+"${ary[@]}"}
              ^-- SC2086: Double quote to prevent globbing and word splitting.
```

#### Here's what I wanted or expected to see:

```
No issues detected!
```

**Comment by koalaman** [comment]
You're right of course. This has been fixed for `+` in 4243c6a0. I'll look at the other cases soon.

**Comment by koalaman** [comment]
The rest was graciously fixed by @martin-schwenke in 807d899. Thanks!
